src/statsd_client.cpp

```diff
@@ -32,217 +37,150 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #include <compat.h>
 #include <netbase.h>
-#include <random.h>
 #include <util/system.h>
 
 #include <cmath>
 #include <cstdio>
+#include <random>
 
 std::unique_ptr<statsd::StatsdClient> g_stats_client;
 
 namespace statsd {
-
-inline bool fequal(float a, float b)
+bool StatsdClient::ShouldSend(float sample_rate)
 {
-    const float epsilon = 0.0001;
-    return ( fabs(a - b) < epsilon );
+    LOCK(cs);
```

Comment:
nit: check m_sock first so we don't do a bunch of logic and then realize we aren't enabled :)

Reply:
Resolved in latest push!